src/App.jsx

```diff
@@ -1,19 +1,37 @@
-import React, { useState } from 'react';
-import TodoList from './TodoList';
-import AddTodoForm from './AddTodoForm';
 import "./App.css";
+import React from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
+
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList')) || []
+  );
+
+  React.useEffect(() => {
+    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList];
+}
 
 function App() {
-  const [todoList, setTodoList] = useState([]); 
+  const [todoList, setTodoList] = useSemiPersistentState();
+
+  const addTodo = (newTodo) => {
+    setTodoList([...todoList, newTodo]);
+  };
 
   return (
     <div>
```

Comment:
This `<div>` element works just fine but take a look at the instructions again you're missing an important component that was brought up in the lesson.

Reply:
thanks I didn't realize, updated